### PR TITLE
Add basic tests for getTestResult()

### PR DESCRIPTION
**Pull request by johannesrohwer**
## Description

This PR adds 3 basic tests for the getTestResult method of the exposure submission service. Particularly relevant is the testGetTestResult_unknownTestResultValue case as discussed in JIRA 1743.

## Checklist

* [x] This PR does not change text that hasn't been signed off with the RKI. Have a look at the pinned issue [440](https://github.com/corona-warn-app/cwa-app-ios/issues)
* [x] Test your changes as thoroughly as possible before you commit them. Preferably, automate your test by unit/integration tests.
* [x] Set a speaking title. Format: {task_name} (closes #{issue_number}). For example: Use logger (closes # 41)
* [x] Make sure that your PR is not introducing _unnecessary_ reformatting (e.g., introduced by on-save hooks in your IDE)

### CODE REVIEW

File: src/xcode/ENA/ENA/Source/Client/__tests__/Mocks/ClientMock.swift [hunk 1]
```diff
@@ -59,7 +63,12 @@ extension ClientMock: Client {
 	}
 
 	func getTestResult(forDevice device: String, completion completeWith: @escaping TestResultHandler) {
-		completeWith(.success(2))
+		guard let onGetTestResult = self.onGetTestResult else {
+			completeWith(.success(2))
```
Comment (haosap) [inline src/xcode/ENA/ENA/Source/Client/__tests__/Mocks/ClientMock.swift:67]: What does the magic number `2 `mean?

File: src/xcode/ENA/ENA/Source/Services/__tests__/ExposureSubmissionServiceTests.swift [hunk 2]
```diff
@@ -142,4 +142,92 @@ class ExposureSubmissionServiceTests: XCTestCase {
...
+
+		let keyRetrieval = MockDiagnosisKeysRetrieval(diagnosisKeysResult: (keys, nil))
+
+		let store = MockTestStore()
+		store.registrationToken = "dummyRegistrationToken"
+
+		let client = ClientMock()
+		client.onGetTestResult = { _, completeWith in
+			let unknownTestResultValue = 4
+			completeWith(.success(unknownTestResultValue))
```
Comment (haosap) [inline src/xcode/ENA/ENA/Source/Services/__tests__/ExposureSubmissionServiceTests.swift:210]: 👍  In this case, I know `unknownTestResultValue`

File: src/xcode/ENA/ENA/Source/Services/__tests__/ExposureSubmissionServiceTests.swift [hunk 3]
```diff
@@ -142,4 +142,92 @@ class ExposureSubmissionServiceTests: XCTestCase {
...
+		waitForExpectations(timeout: .short)
+	}
+
+	func testGetTestResult_noRegistrationToken() {
+
+		// Initialize.
+		let expectation = self.expectation(description: "Expect to receive a result.")
+		let service = ENAExposureSubmissionService(
+			diagnosiskeyRetrieval: MockDiagnosisKeysRetrieval(diagnosisKeysResult: (keys, nil)),
+			client: ClientMock(submissionError: nil),
```
Comment (haosap) [inline src/xcode/ENA/ENA/Source/Services/__tests__/ExposureSubmissionServiceTests.swift:179]: sumissionError has default value `nil`, it's redundant here.

File: src/xcode/ENA/ENA/Source/Services/__tests__/ExposureSubmissionServiceTests.swift [hunk 4]
```diff
@@ -142,4 +142,92 @@ class ExposureSubmissionServiceTests: XCTestCase {
...
+		let expectation = self.expectation(description: "Expect to receive a result.")
+		let expectationToFailWithOther = self.expectation(description: "Expect to fail with error of type .other(_)")
+
+		// Execute test.
+
+		service.getTestResult { result in
+			expectation.fulfill()
+			switch result {
+			case .failure(let error):
+				if case ExposureSubmissionError.other(_) = error {
```
Comment (haosap) [inline src/xcode/ENA/ENA/Source/Services/__tests__/ExposureSubmissionServiceTests.swift:223]: `if case` 👍

**Review by haosap — CHANGES_REQUESTED** [review]

File: src/xcode/ENA/ENA/Source/Client/__tests__/Mocks/ClientMock.swift (hunk 1 above)
Comment (johannesrohwer) [reply src/xcode/ENA/ENA/Source/Client/__tests__/Mocks/ClientMock.swift]: 2 indicates a positive test result. I introduced a new variable to make this more clear.

File: src/xcode/ENA/ENA/Source/Services/__tests__/ExposureSubmissionServiceTests.swift (hunk 3 above)
Comment (johannesrohwer) [reply src/xcode/ENA/ENA/Source/Services/__tests__/ExposureSubmissionServiceTests.swift]: I removed it here and from other places in the file where it was also redundant.

File: src/xcode/ENA/ENA/Source/Client/__tests__/Mocks/ClientMock.swift [hunk 5]
```diff
@@ -19,14 +19,18 @@
...
+	// MARK: - Creating a Mock Client.
+	init(submissionError: SubmissionError? = nil) {
 		self.submissionError = submissionError
 	}
 
-	// MARK: Properties
+	// MARK: - Properties.
 	let submissionError: SubmissionError?
+
+	// MARK: - Configurable Mock Callbacks.
```
Comment (melloskitten) [inline src/xcode/ENA/ENA/Source/Client/__tests__/Mocks/ClientMock.swift:30]: sometimes you have an empty line between MARKs and sometimes you don't. I'm fine with either, just please stick to one.

File: src/xcode/ENA/ENA/Source/Client/__tests__/Mocks/ClientMock.swift [hunk 6]
```diff
@@ -59,7 +63,13 @@ extension ClientMock: Client {
 	}
 
 	func getTestResult(forDevice device: String, completion completeWith: @escaping TestResultHandler) {
-		completeWith(.success(2))
+		guard let onGetTestResult = self.onGetTestResult else {
+			let positiveTestResult = 2
```
Comment (melloskitten) [inline src/xcode/ENA/ENA/Source/Client/__tests__/Mocks/ClientMock.swift:67]: would it perhaps be more clean / easier if you introduced an enum for these test results? you can put it in the top of the class.

**Review by melloskitten — CHANGES_REQUESTED** [review]
Just some minor comments, rest looks good. Thanks for adding tests. cheers 👍

File: src/xcode/ENA/ENA/Source/Client/__tests__/Mocks/ClientMock.swift (hunk 6 above)
Comment (johannesrohwer) [reply src/xcode/ENA/ENA/Source/Client/__tests__/Mocks/ClientMock.swift]: agree, I replaced it.

**Review by melloskitten — APPROVED** [review]
Thanks for adjusting. I'll approve. 💯

**Review by haosap — APPROVED** [review]
nice job! 💯

**Pull request merged**